torch/csrc/jit/codegen/cuda/lower_predicate.cpp

```diff
@@ -75,9 +75,54 @@ class ConditionalFromPredicateModifier : public kir::IrVisitor {
       setWritePredicate(expr, conditional);
     }
 
+    // Note: [Predicate Inversion for CpAsync]
+    // Today for vectorized support the pattern is:
+    // Initialize buffer -> predicated load
+    // For memcpy async:
+    //    If we initialized and then loaded (without sync) it would be undefined
+    //    behavior.
```

Comment:
This is only supported for memcpy async, is that true? It's a little hard for me to tell from the comment exactly where the support works and doesn't, but I think SMEM + MemCpy ASync is the only place this is usable.

Reply:
Yes. This is only done for cp.async, and actually this is the only scenario that I know of where this is required.

In follow ups I think this can be removed for prioritized cases as well even for cp.async, but completely not getting rid of this didn't seem possible.